### DISCUSSION TITLE
Any recommended way to do some stats / monitor?

### DISCUSSION THREAD

**Discussion by williamfzc** [fastapi/fastapi#7403] [Questions]
Hi :)

Currently we have already started an app with fastapi. It looks really good.
Is there a recommended way to do some stats (e.g.: google stats) / monitor (e.g.: [prometheus](https://github.com/prometheus/prometheus) ) about all the requests which we have already received?

Thanks

**Comment by ArcLightSlavik** [comment]
Where do you deploy it? 

Most cloud providers provide some monitoring features which for most people is good enough

This is what i have on GCP:
<img width="1395" alt="Screenshot 2020-08-03 at 13 18 25" src="https://user-images.githubusercontent.com/31127044/89173128-74352880-d58c-11ea-9a14-e36aeb164f4f.png">
<img width="1402" alt="Screenshot 2020-08-03 at 13 18 13" src="https://user-images.githubusercontent.com/31127044/89173136-76978280-d58c-11ea-881c-1ae9db86cebf.png">

**Comment by williamfzc** [comment]
@ArcLightSlavik 
That's what I am looking for. Looks great but we deploy on our private team env without some dashboards like this.
Ok I think this feature should not bind to `fastapi` itself. 
Thanks

**Comment by louisguitton** [comment]
I'm going to shamelessly suggest <a href="https://guitton.co/posts/fastapi-monitoring/">my blog post</a> on the topic of fastapi monitoring 😈 
hopefully you find it interesting

**Comment by williamfzc** [comment]
> I'm going to shamelessly suggest [my blog post](https://guitton.co/posts/fastapi-monitoring/) on the topic of fastapi monitoring 😈
> hopefully you find it interesting

Thanks for your blog and it's helpful :)
But recently I am looking for some private ways so unfortunately we can not use something like datadog
.

**Comment by ArcLightSlavik** [comment]
@louisguitton Looks quite nice, could you make a pull request to add to the list of articles? Would be similar to #2280

**Comment by louisguitton** [comment]
> But recently I am looking for some private ways so unfortunately we can not use something like datadog.

If you're looking for open source solutions, Jaeger is great. the issue is that you won't have a data pipeline to aggregate your traces.
From https://www.jaegertracing.io/roadmap/ :
> Data Pipeline
Post-collection data pipeline for trace aggregation and data mining based on Apache Flink

I'm watching this repo for releases https://github.com/jaegertracing/jaeger-analytics-flink

**Comment by aLowMagic** [comment]
if you want to use the prometheus to monitor this:
1. use docker: you need statsd-exporter to summarizing data then send to prometheus. use datadog's statsd to do that.
2. if just single machine, use prometheus-fastapi-instrumentator

**Comment by itssimon** [comment]
Have a look at [Apitally](https://apitally.io). It's a simple API monitoring & analytics tool that provides insights into API traffic, errors and performance on an intuitive dashboard and easily integrates with FastAPI via a middleware. Also, it never collects potentially sensitive data like request/response bodies or headers, so it should meet your data privacy requirements.

There's a specific [setup guide for FastAPI](https://docs.apitally.io/frameworks/fastapi) too.

*Disclaimer: I'm the founder of Apitally*
